core/src/validator.rs

```diff
@@ -1250,10 +1252,16 @@ impl Validator {
             config.generator_config.clone(),
         );
 
+        let cluster_type = bank_forks.read().unwrap().root_bank().cluster_type();
+        if let Err(e) = metrics_config_sanity_check(cluster_type) {
+            warn!("{e}");
```

Comment:
Not sure if `warn!` is strong enough.  Do we consider failing this unless a specific validator flag is set?

Reply:
I considered this, but wasn't sure if it was too strong of a requirement. Since you also mentioned this I'll go this route.

Reply:
i'd go so hard as to `std::process:exit(1)`. these people have fooled me too many times :wink: